### ISSUE TITLE
Hashrates still wrong?

### ISSUE THREAD

**Issue by Minersworld** [bonesoul/CoiniumServ#980]
Hi,

Have bin playing around with this pool and it seems it's still full of bugs.
Does anyone know how to adjust the hashrates so they are calculated correctly?

Besides this on my front i a few items are in the wrong boxes as like the info from algorithm is under workers, workers is under proffitabilty and so on, so everything needs to be pushed 1 up to the right side.

Thanks.

**Comment by jeong760** [comment]
@Minersworld, Yes I know and will be solved thid weekend. I wil open a PR for this soon

**Comment by Minersworld** [comment]
okey great.
Is this pool still being updated than?
Since i have seen a message about the developers quited this project?

**Comment by bonesoul** [comment]
I'm still out here :)

**Comment by jeong760** [comment]
I am not a programmer but I can able to help for this project

**Comment by Minersworld** [comment]
Okey, than i take back the part about a crappy pool :P
Since that is what i tought... anyway, i purchased a copy to support, so i hope this will be fixed soon,

Of topic, is there any upcomming support for peercoin?

**Comment by jeong760** [comment]
@Minersworld I've requested a PR include a hash rate correction.

**Comment by Minersworld** [comment]
great

Verzonden vanuit Mail voor Windows 10

Van: John Ahn
Verzonden: zondag 11 maart 2018 05:47
Aan: bonesoul/CoiniumServ
CC: Minersworld; Mention
Onderwerp: Re: [bonesoul/CoiniumServ] Hashrates still wrong? (#980)

@Minersworld I've requested a PR include a hash rate correction.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.



---
Deze e-mail is gecontroleerd op virussen door AVG.
http://www.avg.com

**Comment by jeong760** [comment]
@Minersworld, Just refer to the following commit and modify your download repo. because I am still working on other problem to get a IsMine result using getaddressinfo() instead a Validateaddress.

Commit for correction pool hash rate [#9d49d62aaff9813b8ca2948e07a5bc5574b33979](https://github.com/bonesoul/CoiniumServ/pull/982/commits/9d49d62aaff9813b8ca2948e07a5bc5574b33979)

Regards,
John Ahn

**Comment by RepoBike** [comment]
I have a hash rate problem too. 4mos in getting this software working, And now its working. LTC and LCC. 

The only thing I'm having problems is the hash rate note showing.

"Just refer to the following commit and modify your download repo. because I am still working on other problem to get a IsMine result using getaddressinfo() instead a Validateaddress.
Commit for correction pool hash rate #9d49d62aaff9813b8ca2948e07a5bc5574b33979 "

^^^^So how do I put this into the software and fix this error?^^^^

**Comment by jeong760** [comment]
@RepoBike,
 
Have you added a code that I post here? if so, it will displayed  around minute later. 
Or please  post a screenshot you have here..
 
Regards,
John Ahn
-----Original Message-----

**Comment by jeong760** [comment]
@RepoBike, I've corrected an issue to change one word on the code. please change your repo  and re-compile again.
[1ddb53d491db91a4072834b80f165ef6afe3dea7
](https://github.com/bonesoul/CoiniumServ/pull/982/commits/1ddb53d491db91a4072834b80f165ef6afe3dea7)

**Comment by RepoBike** [comment]
Got the hash rate to work. But wow its way off. Miner say 380GH/s and CoiniumServ say 3.84TH/s..

 I wish I could get that from a Antminer S3.. lol.. I'll keep a eye on it and see how it pans out with a 24hr test.

**Comment by jeong760** [comment]
@RepoBike Have you modified a my code  on your repo? if not please clone my repo and re-compile and test it. it should be fixed.

**Comment by RepoBike** [comment]
@jeong760 
```
<Error] [WebServer] [global] Unhandled web-server exception: System.IO.IOException: Unable to write data to the transport connection: Connection reset by peer. ---> System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.Send (System.Byte[] buffer, System.Int32 offset, System.Int32 size, System.Net.Sockets.SocketFlags socketFlags) [0x00016] in <ccafeb0e74bd436bb84e5138772c2bb0>:0
  at System.Net.Sockets.NetworkStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 size) [0x0009b] in <ccafeb0e74bd436bb84e5138772c2bb0>:0
   --- End of inner exception stack trace ---
  at System.Net.Sockets.NetworkStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 size) [0x000e2] in <ccafeb0e74bd436bb84e5138772c2bb0>:0
  at System.Net.ResponseStream.InternalWrite (System.Byte[] buffer, System.Int32 offset, System.Int32 count) [0x0001b] in <ccafeb0e74bd436bb84e5138772c2bb0>:0
  at System.Net.ResponseStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 count) [0x0009c] in <ccafeb0e74bd436bb84e5138772c2bb0>:0
  at System.IO.StreamWriter.Flush (System.Boolean flushStream, System.Boolean flushEncoder) [0x0007e] in <f56c876907e742b0aa586f051fcce845>:0
  at System.IO.StreamWriter.Write (System.String value) [0x00022] in <f56c876907e742b0aa586f051fcce845>:0
  at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.IO.StreamWriter,object)
  at Nancy.ViewEngines.Razor.RazorViewEngine+<>c__DisplayClass2b.<RenderView>b__2a (System.IO.Stream stream) [0x00a4b] in <b530d70cbf1f4cb4993f5ca9450aabf9>:0
  at Nancy.Hosting.Self.NancyHost.OutputWithDefaultTransferEncoding (Nancy.Response nancyResponse, System.Net.HttpListenerResponse response) [0x0000d] in <e8787200efe84e41bf54905d0d49151d>:0
  at Nancy.Hosting.Self.NancyHost.ConvertNancyResponseToResponse (Nancy.Response nancyResponse, System.Net.HttpListenerResponse response) [0x000c8] in <e8787200efe84e41bf54905d0d49151d>:0
  at Nancy.Hosting.Self.NancyHost.Process (System.Net.HttpListenerContext ctx) [0x00027] in <e8787200efe84e41bf54905d0d49151d>:0
19:35:55 [Error] [HybridStorage] [LiteCoin Cash] An exception occurred while committing share: Object reference not set to an instance of an object
19:35:55 [Error] [HybridStorage] [LiteCoin Cash] An exception occurred while committing share: Object reference not set to an instance of an object
19:35:55 [Error] [HybridStorage] [LiteCoin Cash] An exception occurred while committing share: Object reference not set to an instance of an object
19:35:55 [Error] [HybridStorage] [LiteCoin Cash] An exception occurred while committing share: Object reference not set to an instance of an object
19:35:55 [Error] [HybridStorage] [LiteCoin Cash] An exception occurred while committing share: Object reference not set to an instance of an object
19:35:55 [Error] [HybridStorage] [LiteCoin Cash] An exception occurred while committing share: Object reference not set to an instance of an object
19:35:55 [Error] [HybridStorage] [LiteCoin Cash] An exception occurred while committing share: Object reference not set to an instance of an object
19:35:56 [Error] [HybridStorage] [LiteCoin Cash] An exception occurred while committing share: Object reference not set to an instance of an object>
```

After updating today.

**Comment by jeong760** [comment]
@RepoBike  Exactly which algorithm used by Litecoin Cash?

**Comment by RepoBike** [comment]
@jeong760 

LCC- LiteCoin Cash is sha256.

**Comment by mtompkins** [comment]
Hashrate is also off with x16r algo by a similar magnitude. Actual ~50Mh/s shows as 6.17Gh/s on the site

**Comment by jeong760** [comment]
@flyx2888 

Could you check this matter please? I’ve modified a code to get and display coeect hash rate but some people still have an issue.

https://github.com/bonesoul/CoiniumServ/pull/982/commits/9d49d62aaff9813b8ca2948e07a5bc5574b33979

https://github.com/bonesoul/CoiniumServ/pull/982/commits/1ddb53d491db91a4072834b80f165ef6afe3dea7


Regards,
John Ahn

**Comment by flyx2888** [comment]
@jeong760 

Yep, I'll look into it. Show me some screenshots of the problem if you have any.

**Comment by flyx2888** [comment]
@Minersworld 
@RepoBike 
@jeong760 

I'm not seeing these hashrate issues guys. Anyone care to elaborate? Anyways, next I'll be reviewing the code to see if I can spot anything, Will let yall know.

**Comment by flyx2888** [comment]
@Minersworld 
@RepoBike 
@jeong760 

Could someone please post screenshots of where you're seeing the incorrect hashrates? Also specify how long after you connect a new client this is happening and how many clients? I'm currently debugging this problem and may have to correct some code.

**Comment by jeong760** [comment]
@flyx2888  After I've changed a code and applied I have no issue to display a pool hash rate but other will have an issue.

**Comment by flyx2888** [comment]
@jeong760 

Okay, thank you. I will run some further tests to see what's going on here.

**Comment by flyx2888** [comment]
After thorough testing, I've concluded that the current version version is reporting hashrates correctly. Going back several versions, each had problems of over reporting the hashrates.

There may be other factors involved here as to why some people are getting wrong or non-existent hashrates. We'll have to look at each on an individual basis to determine possible causes.

Anyone still with problems, state your case and we'll look into it.

**Comment by Mentors4EDU** [comment]
Still having this issue: #1071
